src/shared/logs-show.c

```diff
@@ -578,6 +632,37 @@ static int output_short(
                                 fputc(' ', f);
                         }
                 }
+
+                if (build_id && shall_print_refuse_nul(build_id, build_id_len, flags) &&
```

Comment:
I think this must be opt-in, similarly to how we print locations. It is definitely a useful feature, but it will only be used in 0.000001% of cases. In particular, you'll only use it when debugging a program. Even if you are a developer, the logs contain messages from thousands of different programs, and you are never going to look at any of them, except for the few programs that you care about. So if we multiply the fraction of people who are able and want to read source code times the fraction of people who are currently doing debugging times the fraction of programs you are interested in, it's hard to justify turning it on by default.